drivers/sensor/stm32_vref/Kconfig

```diff
@@ -0,0 +1,12 @@
+# STM32 vref sensor configuration options
+
+# Copyright (c) 2023 Kenneth J. Miller <ken@miller.ec>
+# SPDX-License-Identifier: Apache-2.0
+
+config STM32_VREF
+	bool "STM32 VREF Sensor"
+	default y
+	depends on DT_HAS_ST_STM32_VREF_ENABLED
+	depends on (ADC && SOC_FAMILY_STM32)
```

Comment:
```suggestion
	select ADC
```

Reply:
This is more concise and seems to be in line with the structure of most other driver Kconfigs.

Amended original commit, thanks!